exporter/redis.go

```diff
@@ -52,6 +55,70 @@ func (e *Exporter) connectToRedis() (redis.Conn, error) {
 	return c, err
 }
 
+func (e *Exporter) connectToRedisCluster() (redis.Conn, error) {
+	tlsConfig, err := e.CreateClientTLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	options := []redis.DialOption{
+		redis.DialConnectTimeout(e.options.ConnectionTimeouts),
+		redis.DialReadTimeout(e.options.ConnectionTimeouts),
+		redis.DialWriteTimeout(e.options.ConnectionTimeouts),
+		redis.DialTLSConfig(tlsConfig),
+	}
+
+	if e.options.User != "" {
+		options = append(options, redis.DialUsername(e.options.User))
+	}
+
+	if e.options.Password != "" {
+		options = append(options, redis.DialPassword(e.options.Password))
+	}
+
+	uri := e.redisAddr
+	if strings.Contains(uri, "://") {
+		url, _ := url.Parse(uri)
+		if url.Port() == "" {
+			uri = url.Host + ":6379"
+		} else {
+			uri = url.Host
+		}
+	} else {
+		if frags := strings.Split(uri, ":"); len(frags) != 2 {
+			uri = uri + ":6379"
+		}
+	}
+	log.Debugf("Trying DialURL(): %s", uri)
+
+	if e.options.PasswordMap[uri] != "" {
+		options = append(options, redis.DialPassword(e.options.PasswordMap[uri]))
+	}
```

Comment:
There's a lot of copy-paste from `connectToRedis()` - consider moving lines 19-43 into a separate function and reuse it here and in connecttoRedis()

Reply:
This still applies - I think you could consolidate a bit here.